src/c3chart/CommonND.js

```diff
@@ -16,7 +16,7 @@
         };
 
         this._config.data.onclick = function (d, element) {
-            context.click(context.rowToObj(context._data[d.index]), d.id);
+            context.click(context.rowToObj(context.data()[d.index]), d.id);
```

Comment:
You need to pass in a third param which is true/false depending on if the click selects or de-selects (which is what Matt is pointing out)